src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrBackflower.cs

```diff
@@ -506,7 +506,7 @@ .. mapping.Exclude.Select(VmrPatchHandler.GetExclusionRule),
         string targetBranch,
         CancellationToken cancellationToken)
     {
-        await _vmrCloneManager.PrepareVmrAsync([build.GetRepository()], [build.Commit], build.Commit, cancellationToken);
+        await _vmrCloneManager.PrepareVmrAsync(build.GetRepository(), [build.Commit], build.Commit, ShouldResetBranchToRemoteWhenPreparingVmr(), cancellationToken);
```

Comment:
So here we know that this is going to be a SHA. Should we still reset?
What will actually happen?

Reply:
I don't think we have to. We will fetch that commit when we're calling `git remote update`, and then we will just check it out. Since we're checking out the exact commit here, and not the branch, it should always match with the remote

Reply:
I think I like having a property for it, and having a comment describing why it's set to it. For example, during backflow, we always want to pass false when cloning the VMR, because we're checking out a specific commit the build was made from, not the branch